src/scan/postgres_scan.cpp

```diff
@@ -11,6 +11,12 @@
 
 #include <numeric> // std::accumulate
 
+extern "C" {
+#include "postgres.h"
+
+#include "utils/memutils.h"
+}
+
```

Comment:
Instead of inlcuding postgres headers here (in a file that we managed to remove all postgres includes from). Let's create some basic wrapper functions in a new file in `src/pg/`, maybe called `memory.cpp`. Also `AllocSetContextCreate` and `MemoryContextReset` need to be wrapped in a `PostgresFunctionGuard` when doing so.

Reply:
good point!